docs/src/catalyst_applications/jump_simulation_performance.md

```diff
@@ -0,0 +1,163 @@
+
+#### Using piecewise constant, time-dependant, function instead of callbacks.
+Let us consider a reaction with some input that depends on time. We assume that the input is initially $0$, but after some critical time $t=5.0$ it is increased to $1$. This can be implemented through a step function:
+```@example jump_simulation_performance_1
+input(t) = t > 5.0
+r = @reaction input(t), 0 --> X
+nothing # hide
+```
+Here, the production of species $X$ is switched on at the time $t=5.0$. This reaction (which rate depends on `t`) will generate a `VariableRateJump`, which is bad for jump simulation performance. However, since the rate is piecewise constant, it can instead be implemented by setting it to a constant parameter $i$, and then use a [*callback*](@ref advanced_simulations_callbacks) to update it at the critical times. This will again generate an equivalent model, but with the reaction encoded as a `MassActionJump` (rather than a `VariableRateJump`).
+
+## Jump solver selection
+When creating a `JumpProblem`, a specific solver is designated using its third argument.
+```@example jump_simulation_performance_2
+using Catalyst, JumpProcesses, Plots
+
+rn = @reaction_network begin
+    (p,d), 0 <--> X
+end
+u0 = [:X => 10]
+tspan = (0.0, 1000.0)
+ps = [:p => 1000.0, :d => 100.0]
+
+dprob = DiscreteProblem(rn, u0, tspan, ps)
+jprob = JumpProblem(rn, dprob, Direct())
+nothing # hide
+```
+Here (as throughout most of Catalyst's documentation) we have used the `Direct()` solver (which corresponds to Gillespie's original direct method [^2][^3], also called the *stochastic simulation algorithm*). This method was originally published in 1976, and since then, many additional methods for performing jump simulations of CRN models have been developed. 
+
+Gillespie's direct method will, after a jump has been performed, recompute the rates of *all* possible jumps in the system. This is typically not required. E.g. consider the following system:
+```@example jump_simulation_performance_2
+@reaction_network begin
+    k1, X1 --> X2
+    k2, X2 --> X3
+    k3, X3 --> 0
+end
+```
+Here, the rate of the `k1, X1 --> X2` and `k2, X2 --> X3` reactions does not depend on the amount of $X3$ in the system. Hence, their rates are unaffected by the occurrence of the `k3, X3 --> 0` reaction. Performant jump simulation methods have clever ways to determine which rates require recomputing after the occurrence of each reaction, which improves their performance. Many of these depend on so-called dependency graphs (which track which reactions' rates are affected by the occurrence of which reactions). Catalyst automatically builds such dependency graphs, which means that most jump simulators can be used without any additional input.
```

Comment:
```suggestion
Here, the propensities of the `k1, X1 --> X2` and `k2, X2 --> X3` reactions, `k1*X1*X2` and `k2*X2` respectively, do not depend on the amount of $X3$ in the system. Hence, the value of their propensities are unchanged by the occurrence of the `k3, X3 --> 0` reaction. Performant jump simulation methods have several ways to determine which rates require recomputing after the occurrence of a reaction, which improves their computational performance. Many of these depend on dependency graphs, which track which other reactions' propensities must be recomputed after the occurrence of a given reaction. Catalyst automatically builds such dependency graphs, which means that all JumpProcesses SSAs can be used without any additional inputs.
```

Note, one needs to be careful about using rate in place of propensity. In Catalyst rate usually refers to the `rate` field of a `Reaction`, which is not the propensity function (the latter is more the stochastic rate law).

Reply:
Yes. I was trying to keep things simple by referring to only rates, rather than propensities and rates, The idea was to make it easier for someone who hasn't studied the theory and wasn't familiar with these terms. We can rewrite it to use both, but in that case I might want a more throughout introduction of propensities at the earlier stage.